### PR TITLE
Throw on invalid histogram bounds

### LINKED ISSUE #2566: Histogram bounds must be validated
Similar to instrument/view name, we need to validate the custom bounds (no duplicates, increasing order etc.), if provided.
If we can fail-fast, do that. - done in #2573 
Else log and move on.

### PR DESCRIPTION
**Pull request by mic-max**
Fixes #2566

## Changes

Throws `ArgumentException` when invalid histogram bounds are provided.
The bounds must be in ascending order and contain distinct values.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Tests/Metrics/MetricsTestData.cs [hunk 1]
```diff
@@ -40,5 +40,12 @@ public static IEnumerable<object[]> ValidInstrumentNames
                     new object[] { "my_metric2" },
                     new object[] { new string('m', 63) },
            };
+
+        public static IEnumerable<object[]> InvalidHistogramBounds
+           => new List<object[]>
+           {
+                    new object[] { new double[] { 0, 0 } },
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Metrics/MetricsTestData.cs:47]: perhaps add a couple more edge cases.

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]
